core/src/main/java/io/grpc/internal/RetryingNameResolver.java

```diff
@@ -95,12 +95,26 @@ public void onResult(ResolutionResult resolutionResult) {
             "RetryingNameResolver can only be used once to wrap a NameResolver");
       }
 
+      // To have retry behavior for name resolvers that haven't migrated to onResult2.
       delegateListener.onResult(resolutionResult.toBuilder().setAttributes(
               resolutionResult.getAttributes().toBuilder()
                   .set(RESOLUTION_RESULT_LISTENER_KEY, new ResolutionResultListener()).build())
           .build());
     }
 
+    @Override
+    public Status onResult2(ResolutionResult resolutionResult) {
+      Status status = delegateListener.onResult2(resolutionResult);
+      if (status != null) { // null indicates LB had been shut down.
```

Comment:
We don't want null return values. And if null has special meaning, that would need to be documented in onResult2's javadoc.

Reply:
Removed this condition as we are not returning null anywhere. I wrote this in an earlier attempt where I was returning null instead of FAILED_PRECONDITION.